lib/components/normal-components/Resonator.ts

```diff
@@ -0,0 +1,47 @@
+import { resonatorProps } from "@tscircuit/props"
+import { NormalComponent } from "../base-components/NormalComponent"
+import type { BaseSymbolName } from "lib/utils/constants"
+
+export class Resonator extends NormalComponent<typeof resonatorProps> {
+  get config() {
+    const getSymbolName = (variant: string | undefined): BaseSymbolName => {
```

Comment:
don't nest functions unnecessarily, move the function outside of the `get config() {` function

why? it simplifies nesting code and allows you to use multiple files.

You should rename the function `getResonatorSymbolName` but you can keep it in this file since it's already so short

Reply:
CC @Abse2001 this is a good lint rule to add, this pattern isn't allowed on any project

I think eslint has a version of it although we'll want to be careful not to break e.g. react

https://eslint.org/docs/latest/rules/no-inner-declarations

![image](https://github.com/user-attachments/assets/12872a46-b16c-4111-80f3-91dfc7f7a8f8)

Reply:
done.